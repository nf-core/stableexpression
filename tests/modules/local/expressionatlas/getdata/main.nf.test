nextflow_process {

    name "Test Process EXPRESSIONATLAS_GETDATA"
    script "modules/local/expressionatlas/getdata/main.nf"
    process "EXPRESSIONATLAS_GETDATA"
    tag "getdata"
    tag "module"

    test("Transcriptome Analysis of the potato (rnaseq)") {

        tag "getdata_potato"

        when {

            process {
                """
                input[0] = "E-MTAB-552"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Transcription profiling by array of Arabidopsis mutant for fis2 (microarray)") {

        tag "getdata_arabido"

        when {

            process {
                """
                input[0] = "E-TABM-1007"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Arabidopsis Geo dataset") {

        tag "getdata_arabido_geo"

        when {

            process {
                """
                input[0] = "E-GEOD-62537"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Invalid accession") {

        tag "getdata_invalid"

        when {

            process {
                """
                input[0] = "fake-accession"
                """
            }
        }

        then {
            assert process.success
            assert process.trace.succeeded().size() == 0
            assert process.trace.failed().size() == 1
            // check for the absence of expected output (the error is ignored but no output is produced)
            assert process.out.design.size() == 0

        }

    }

    test("Accession not available") {

        tag "getdata_unavailable"

        when {

            process {
                """
                input[0] = "E-GEOD-161565656"
                """
            }
        }

        then {
            assert process.success
            assert process.trace.succeeded().size() == 0
            assert process.trace.failed().size() == 1
            // check for the absence of expected output (the error is ignored but no output is produced)
            assert process.out.design.size() == 0

        }

    }

    test("E-MTAB-5132") {

        tag "getdata_unavailable"

        when {

            process {
                """
                input[0] = "E-MTAB-5132"
                """
            }
        }

        then {
            assert process.success
            assert process.trace.succeeded().size() == 0
            assert process.trace.failed().size() == 1
            // check for the absence of expected output (the error is ignored but no output is produced)
            assert process.out.design.size() == 0
        }

    }

    test("E-PROT-138") {

        tag "getdata_unsupported"

        when {

            process {
                """
                input[0] = "E-PROT-138"
                """
            }
        }

        then {
            assert process.success
            assert process.trace.succeeded().size() == 0
            assert process.trace.failed().size() == 1
            // check for the absence of expected output (the error is ignored but no output is produced)
            assert process.out.design.size() == 0
        }

    }

}
