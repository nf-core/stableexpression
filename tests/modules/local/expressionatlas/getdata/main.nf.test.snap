{
    "Transcriptome Analysis of the potato (rnaseq)": {
        "content": [
            {
                "0": [
                    "E_MTAB_552_rnaseq.design.csv:md5,b81490696f638e90c1cf14236bb0c08c"
                ],
                "1": [
                    "E_MTAB_552_rnaseq.raw.csv:md5,830f50b60b17b62f9ca2f6a163a2879f"
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "1.30.0"
                    ]
                ],
                "design": [
                    "E_MTAB_552_rnaseq.design.csv:md5,b81490696f638e90c1cf14236bb0c08c"
                ],
                "normalised": [
                    
                ],
                "raw": [
                    "E_MTAB_552_rnaseq.raw.csv:md5,830f50b60b17b62f9ca2f6a163a2879f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-04T11:40:48.88652487"
    },
    "Arabidopsis Geo dataset": {
        "content": [
            {
                "0": [
                    "E_GEOD_62537_A_AFFY_2.design.csv:md5,7d7dd72be4f5b326dd25a36db01ebf88"
                ],
                "1": [
                    
                ],
                "2": [
                    "E_GEOD_62537_A_AFFY_2.normalised.csv:md5,673c55171d0ccfc1d036bf43c49ae320"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "1.30.0"
                    ]
                ],
                "design": [
                    "E_GEOD_62537_A_AFFY_2.design.csv:md5,7d7dd72be4f5b326dd25a36db01ebf88"
                ],
                "normalised": [
                    "E_GEOD_62537_A_AFFY_2.normalised.csv:md5,673c55171d0ccfc1d036bf43c49ae320"
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-04T11:41:28.635715975"
    },
    "Transcription profiling by array of Arabidopsis mutant for fis2 (microarray)": {
        "content": [
            {
                "0": [
                    "E_TABM_1007_A_AFFY_2.design.csv:md5,120f63cae2193b97d7451483bdbbaab1"
                ],
                "1": [
                    
                ],
                "2": [
                    "E_TABM_1007_A_AFFY_2.normalised.csv:md5,a3afe33d7eaed3339da9109bf25bb3ed"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "1.30.0"
                    ]
                ],
                "design": [
                    "E_TABM_1007_A_AFFY_2.design.csv:md5,120f63cae2193b97d7451483bdbbaab1"
                ],
                "normalised": [
                    "E_TABM_1007_A_AFFY_2.normalised.csv:md5,a3afe33d7eaed3339da9109bf25bb3ed"
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-04T11:41:08.294214693"
    }
}